plugins/remote/enroll/tls_enroll.cpp

```diff
@@ -73,23 +78,43 @@ std::string TLSEnrollPlugin::enroll() {
   }
 
   std::string node_key;
-  bool should_shutdown = false;
   VLOG(1) << "TLSEnrollPlugin requesting a node enroll key from: " << uri;
-  for (size_t i = 1; i <= FLAGS_tls_enroll_max_attempts && !should_shutdown;
-       i++) {
+
+  std::uint64_t attempt = 1;
+  std::uint64_t interval_step = 1;
+  std::uint64_t interval = 0;
+
+  bool should_shutdown = false;
+
+  for (;;) {
     auto status = requestKey(uri, node_key);
-    if (status.ok() || i == FLAGS_tls_enroll_max_attempts) {
+    if (status.ok() || attempt == FLAGS_tls_enroll_max_attempts) {
```

Comment:
What happens if `FLAGS_tls_enroll_max_attempts` is set to 1?

Reply:
It will exit after 1 request, not matter if failed or succeeded.
This is the same of the previous behavior, where the number of attempts means the total attempts done, not the times it will do a retry (after a first failure).

Reply:
@seph Updated, I've used the first sentence of the wiki doc that it seemed cleared to me.
Does it make sense now?

Reply:
Oh, it wasn't a doc question. I'm being somewhat unclear on the code. I see it now -- the request is line 91

I assume the `==` is fine, but I always fear somehow attempt will end up greater than max, and it'll loop forever.

Reply:
I see, I've changed to use a while loop with the exit conditions explicit (although they are redundant).

Reply:
You could strike the redundancy here, or leave it. I'm generally happy to accept that I'm slower to understand c++